cmd/registry/breaking-change-detector/breaking_change_detector.go

```diff
@@ -0,0 +1,237 @@
+package breakingchangedetector
+
+import (
+	"regexp"
+	"strconv"
+
+	"github.com/apigee/registry/rpc"
+)
+
+type detectionTypeList struct {
+	breakingChanges    detectionList
+	nonBreakingChanges detectionList
+}
+type detectionList struct {
+	additions     []detectionPattern
+	deletions     []detectionPattern
+	modifications []detectionPattern
+}
+
+type detectionPattern struct {
+	detectionWeight    int
+	PositiveMatchRegex *regexp.Regexp
+	NegativeMatchRegex *regexp.Regexp
+}
+
+// GetBreakingChanges compares each change in a diff Proto to the relavant change type detection Patterns.
+// Each change is then catgorized as breaking, nonbreaking, or unknown.
+func GetBreakingChanges(diff *rpc.Diff) *rpc.ClassifiedChanges {
+	changePatterns := getChangePatterns()
+	changesProto := compareChangesToPatterns(changePatterns, diff)
+	return changesProto
+}
+
+// getChangePatterns Intalizes the Breaking and NonBreaking change Patterns.
+func getChangePatterns() detectionTypeList {
```

Comment:
This function seems like it should be a variable. Looking ahead, it might be best as several variables that aren't contained in a `detectionList` struct.

I see this function being called once, then the result being passed through a few other functions as parameters, then its fields being accessed once each. Instead we can define variables for `d.breakingChanges.additions`, `d.nonBreakingChanges.additions`, `d.breakingChanges.deletions`, `d.nonBreakingChanges.deletions`, `d.breakingChanges.modifications`, and `d.nonBreakingChanges.modifications`.

Something like the following.
```
var (
  unsafeAdds = []regexp.Regexp{ ... }
  unsafeDeletes = []regexp.Regexp{ ... }
  unsafeMods = []regexp.Regexp{ ... }
  safeAdds = []regexp.Regexp{ ... }
  safeDeletes = []regexp.Regexp{ ... }
  safeMods = []regexp.Regexp{ ... }
)
```

Reply:
Done